### PR TITLE
[Instrumentation.Runtime] Reformat by taking description column out of table

### PR DESCRIPTION
**Pull request by xiang17**
Rearranging the README.md to form headings for each metric. The original format is from the table in [JVM metrics](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#jvm-metrics), but the description inside table is too long and not friendly to read.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #335

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 1]
```diff
@@ -64,9 +64,13 @@ complete demo.
...
 ### GC related metrics
 
-| Name                                            | Description                                                           | Units           | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
-|-------------------------------------------------|-----------------------------------------------------------------------|-----------------|-------------------|------------|------------------|------------------|
-| process.runtime.dotnet.**gc.collections.count** | Number of garbage collections that have occurred since process start. | `{collections}` | ObservableCounter | `Int64`    | generation       | gen0, gen1, gen2 |
+#### process.runtime.dotnet.**gc.collections.count**
+
+| Name                                            | Units           | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|-------------------------------------------------|-----------------|-------------------|------------|------------------|------------------|
+| process.runtime.dotnet.**gc.collections.count** | `{collections}` | ObservableCounter | `Int64`    | generation       | gen0, gen1, gen2 |
```
[the remark this answers is not in the document]
Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:71]: agree

**Review by cijothomas — COMMENTED** [review]
Like this approach. Its now easy to link to a specific metric in the doc. (I see a lot of needs to have the links easily available!)

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 2]
```diff
@@ -79,50 +83,113 @@ The API used to retrieve the value is:
   The number of times garbage collection has occurred for the specified generation
 of objects.
 
-#### Additional GC metrics only available when targeting .NET Core 3.1 or later
+#### process.runtime.dotnet.**gc.allocations.size**
+
+| Units   | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:88]: there is no attributes, so maybe make it clear that this metric has no dimensions?

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:88]: Updated to `| No Attributes    | N/A              |`.

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 3]
```diff
@@ -64,9 +64,13 @@ complete demo.
...
-| Name                                            | Description                                                           | Units           | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
-|-------------------------------------------------|-----------------------------------------------------------------------|-----------------|-------------------|------------|------------------|------------------|
-| process.runtime.dotnet.**gc.collections.count** | Number of garbage collections that have occurred since process start. | `{collections}` | ObservableCounter | `Int64`    | generation       | gen0, gen1, gen2 |
+#### process.runtime.dotnet.**gc.collections.count**
+
+| Units           | Instrument Type   | Value Type | Attribute Key(s) | Attribute Values |
+|-----------------|-------------------|------------|------------------|------------------|
+| `{collections}` | ObservableCounter | `Int64`    | generation       | gen0, gen1, gen2 |
+
+Description: Number of garbage collections that have occurred since process start.
```
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:73]: I suggest that we make the description the 1st sentence of each metric.

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:69]: Done.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Runtime/README.md (hunk 3 above)
Comment (reyang) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:69]: @xiang17 could you also remove the "Description: " part? It doesn't seem to add value here.

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**